File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/HowTo/GetMetadata.cs
using Raven.Client.Document;
using Raven.Documentation.CodeSamples.Orders;
using Raven.Json.Linq;

namespace Raven.Documentation.Samples.ClientApi.Session.HowTo
{
	public class GetMetadata
	{
		private interface IFoo
		{
			#region get_metadata_1
			RavenJObject GetMetadataFor<T>(T instance);
			#endregion
		}

		public GetMetadata()
		{
			using (var store = new DocumentStore())
			{
				using (var session = store.OpenSession())
				{
					#region get_metadata_2
					Employee employee = session.Load<Employee>("employees/1");
					RavenJObject metadata = session.Advanced.GetMetadataFor(employee);
					#endregion
				}
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/session/howto/GetMetadata.java
package net.ravendb.clientapi.session.howto;

import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentSession;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.samples.northwind.Employee;


public class GetMetadata {

  @SuppressWarnings("unused")
  private interface IFoo {
    //region get_metadata_1
    public <T> RavenJObject getMetadataFor(T instance);
    //endregion
  }

  @SuppressWarnings("unused")
  public GetMetadata() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      try (IDocumentSession session = store.openSession()) {
        //region get_metadata_2
        Employee employee = session.load(Employee.class, "employees/1");
        RavenJObject metadata = session.advanced().getMetadataFor(employee);
        //endregion
      }
    }
  }
}
